docs(readme, contributing): add contribution guidelines (#192)
ShabadOS/desktop: CONTRIBUTING.md

# Contributing Guildeines

We're glad you're interesting in contributing!

## Style

This repository follows the **Airbnb's Javascript Style Guide**, with a few minor modifications. Notably, spaces should be included inside parentheses and brackets (weird, right!).

An ESLint file is provided,
and your code will automatically be checked on-commit for style.
It is recommended to install an ESLint plugin for your editor (VS Code's `ESLint` plugin works out of the box), so you can receive linter suggestions as you type.

Create one feature per branch, preferably working against an issue. PRs will be squashed and merged when complete.

## Commit Message Guidelines

Each commit message consists of a header, a body and a footer. The header has a special format that includes a type, a scope and a subject:

```
<type>(<scope>): <subject>
<BLANK LINE>
<body>
<BLANK LINE>
<footer>
````

The header is mandatory and the scope of the header is optional.

Any line of the commit message cannot be longer 100 characters! This allows the message to be easier to read on GitHub as well as in various git tools.

> The footer should contain a closing reference to an issue if any, such as `Closes #21`.

Reviewer: ```suggestion
The footer should contain references to any issues, such as `Closes #21` or `Related #75`.
```

Examples:
```
docs(readme): add contributing guidelines
```
```
fix(backend): synchronise state with frontend

The backend would not send all required data to the frontend.

Fixes #53
```

### Revert

> If the commit reverts a previous commit, it should begin with `revert:` , followed by the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.

Reviewer: an example would help. should it be: `revert: docs(readme): add contributing guidelines` or `revert: add contributing guidelines` or some combo of quotes etc.

Author: The former, i'll add the example

### Type

Must be one of the following:

- *build*: Changes that affect the build system or external dependencies (example scopes:  -gulp, broccoli, npm)
- *ci*: Changes to our CI configuration files and scripts (example scopes: Circle,  -BrowserStack, SauceLabs)
- *docs*: Documentation only changes
- *feat*: A new feature

> - *fix*: A bug fix

Reviewer: currently we use "fix" for correcting gurbani in db repo. ideally these changes to content should have it's own "type" since they are not fixes, refactors, or style changes. for instance a type called `typo` might be nice. what would we categorize https://github.com/ShabadOS/desktop/issues/186 ? not a bug to fix, not a "code" change per se, and not stylistic changes to the "code". Unless we consider content to be code? In which case we should use "refactor" for db commits and #186 ...?

Author: #186 would either be a `refactor` or `feat` I reckon. I think `fix` is still fine for DB commits, since we're still fixing an issue, right?

- *perf*: A code change that improves performance
- *refactor*: A code change that neither fixes a bug nor adds a feature
- *style*: Changes that do not affect the meaning of the code (white-space, formatting,  -missing semi-colons, etc)
- *test*: Adding missing tests or correcting existing tests

### Scope

The scope should be the name of the npm package affected (as perceived by the person reading the changelog generated from commit messages.

The following is the list of supported scopes:

backend
frontend
frontend/controller
frontend/overlay
frontend/presenter
frontend/settings
frontend/themes

There are currently a few exceptions to the "use package name" rule:

> - *packaging*: used for changes that change the npm package layout in all of our packages, e.g. public path changes, package.json changes done to all packages, d.ts file/format changes, changes to bundles, etc.

Reviewer: ```suggestion
- *packaging*: used for changes to the npm package layout in all of our packages, e.g. public path changes, package.json changes done to all packages, d.ts file/format changes, changes to bundles, etc.
```
